### PR TITLE
Add `module: es2022`

### LINKED ISSUE #44653: Add `module: es2022`
# Suggestion

## 🔍 Search Terms

es2022

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Add `module: es2022` and allow top level await in it.

## 📃 Motivating Example

Top level await

## 💻 Use Cases

Not being required to set esnext.

Related to #44571.

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #44653

### CODE REVIEW

File: src/testRunner/unittests/config/convertCompilerOptionsFromJson.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ namespace ts {
                         file: undefined,
                         start: 0,
                         length: 0,
-                        messageText: "Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'esnext'.",
+                        messageText: "Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'es2020', 'es2022', 'esnext'.",
```
Comment (saschanaz) [inline src/testRunner/unittests/config/convertCompilerOptionsFromJson.ts:187]: Interesting to see this hasn't failed. I don't think this even runs 🤔

**Comment by saschanaz** [comment]
Do we want this function to return ModuleKind.ES2020 and ES2022 following the `target` version? It seems #33893 missed this, and I wonder we want to fix it.

https://github.com/microsoft/TypeScript/blob/5afe42e14e61d7e4df5d75cc0022283711cb593a/src/compiler/utilities.ts#L6035-L6039

**Comment by MartinJohns** [comment]
PR seems kind of outdated.

**Comment by saschanaz** [comment]
I'll rebase once anyone reviews this.

**Comment by weswigham** [comment]
This is going to look a weeee bit different after a merge since the module: node changes touch many of the same lines, but initially it looks ok.

**Comment by simllll** [comment]
Next to module: node12, this is another thing I'm really looking forward to.

**Review by weswigham — APPROVED** [review]
This little looks fine to me - @DanielRosenwasser do we want this in for the beta?

**Review by DanielRosenwasser — APPROVED** [review]
I think this seems reasonable.

**Comment by stanflows** [comment]
What could be the default option for this and the highest version, I guess is none and esnext?

**Pull request merged**